ConfidentialClientTokenCache/Program.cs

```diff
@@ -64,8 +49,46 @@ static async Task Main(string[] args)
                 .WithTenantId(tenant)
                 .Build();
 
-            msalTokenCacheProvider.Initialize(app.UserTokenCache);
-            msalTokenCacheProvider.Initialize(app.AppTokenCache);
+            // Distributed in memory token caches (App and User caches)
+            app.AddInMemoryTokenCaches();
```

Comment:
maybe we'd want to use a #define 
Something like:
#define SERIALIZATION_IMPLEMENTATION InMemory // Or DistributedInMemory, Redis, Sql, CosmosDb

then:

```CSharp
#if SERIALIZATION_IMPLEMENTATION InMemory
            // Distributed in memory token caches (App and User caches)
            app.AddInMemoryTokenCaches();
#endif 

#if DistributedInMemory
             // Distributed token caches (App and User caches)
            // Add one of the below: SQL, Redis, CosmosDb
            app.AddDistributedTokenCaches(services =>
            {
             services.AddDistributedInMemoryCache();
            });
#endif 

```

etc ...

Reply:
some people don't know what #define means so i'll just comment it.